### PR TITLE
Solutions for P01-P07

### PR DESCRIPTION
**Pull request by alexmarchis1990**

### CODE REVIEW

File: Solutions/alexmarchis1990/p06_alexmarchis1990.swift [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import NinetyNineSwiftProblems
+
+extension List where T:Equatable {
+    
+    func isPalindrome() -> Bool {
+        §
```
Comment (eneko) [inline Solutions/alexmarchis1990/p06_alexmarchis1990.swift]: There is a weird `§` character here.

**Comment by alexmarchis1990** [comment]
Hi @eneko 
Can you give me a hint on what's the issue with travis? It still fails even after I've pushed the sources.

**Comment by eneko** [comment]
Going to restart the build, not sure why it failed. Seems like a permissions issue, I might need to check how Danger works with forked repositories.

File: Solutions/alexmarchis1990/p05_alexmarchis1990.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+        var arr:[T] = []
+        arr.append(self.value)
+        var currentNode = self.nextItem
+        
+        while let cn = currentNode {
+            arr.insert(cn.value, at: 0)
+            currentNode = cn.nextItem
+        }
+        
+        return List<T>.init(arr)!
```
Comment (eneko) [inline Solutions/alexmarchis1990/p05_alexmarchis1990.swift]: Using `Array` or any other sequence types from the standard library is not allowed, as it defeats the purpose of these problems.

**Comment by eneko** [comment]
Hi @alexmarchis1990,

Seems like the build is failing now because the `List` initializers no longer return optional. Please see if you can fix that. Regarding the warnings, please ignore those, I was doing some tests with SwiftLint.

**Comment by alexmarchis1990** [comment]
Hi @eneko 
I've made the required corrections and also added a solution for P07. Let me know if any problems, thanks!

**Comment by eneko** [comment]
Awesome, good job!

**Pull request merged**